src/shredder/reed_solomon.rs

```diff
@@ -111,7 +112,15 @@ pub(super) fn reed_solomon_deshred(
     }
 
     // remove padding
-    let padding_bytes = restored_payload[restored_payload.len() - 1] as usize;
+    let padding_bytes = restored_payload
```

Comment:
I guess the downside of this approach is that in the worse case we have to check the last `2 * DATA_SHREDS` bytes to find the the padding.  Currently `DATA_SHREDS` is 32 so shouldn't be very expensive but could become more expensive as we increase the number.  We can worry about that later probably.

Reply:
yeah, we could think about making that change in the future.  I'm happy to keep things as is in this PR.